dlt/destinations/dataset/factory.py

```diff
@@ -0,0 +1,22 @@
+from typing import Union
+
+
+from dlt.common.destination import AnyDestination
+from dlt.common.destination.reference import (
+    SupportsReadableDataset,
+    TDatasetType,
+    TDestinationReferenceArg,
+)
+
+from dlt.common.schema import Schema
+
+from dlt.destinations.dataset.dataset import ReadableDBAPIDataset
+
+
+def dataset(
```

Comment:
do you want to improve typing in this PR? if so we could
1. make `SupportsReadableDataset` generic where it is parametrized by relation type (bound to SupportsReadableRelation)
2. use overload on the literal `TDatasetType` when in auto/ibis you return SupportsReadableDataset parametrized by ReadableIbisRelation, otherwise by just SupportsReadableRelation
3. we can make (2) dynamic dependent if ibis is present

Reply:
Can you maybe explain how (3) works? I don't understand how mypy can lint something else if a certain library is installed..

Reply:
Anyway as mentioned above, I have now brute forced typing, but i'm happy to improve this if I understand how to do (3)